### PR TITLE
Simplify finalizing native reports

### PR DESCRIPTION
**Pull request by mrwillis21**
Collapse finalizing previous native report into existing session
finalize operation.

### CODE REVIEW

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java [hunk 1]
```diff
@@ -858,6 +858,9 @@ private void doCloseSessions(int maxCustomExceptionEvents, boolean includeCurren
 
     if (includeCurrent) {
       reportingCoordinator.onEndSession();
+    } else if (nativeComponent.hasCrashDataForSession(mostRecentSessionIdToClose)) {
```
Comment (jakeouellette) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java]: can you add some justification for why this is an else if? why do we only do native stuff if includeCurrent = false? Why do we not do it if includeCurrent = false?

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java [hunk 2]
```diff
@@ -858,6 +858,9 @@ private void doCloseSessions(int maxCustomExceptionEvents, boolean includeCurren
...
+      finalizePreviousNativeSession(mostRecentSessionIdToClose);
+      nativeComponent.finalizeSession(mostRecentSessionIdToClose);
```
Comment (jakeouellette) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java]: before we were returning a boolean and logging based on it, do we need to do an equivalent?

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java (hunk 1 above)
Comment (mrwillis21) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java]: Yeah, this is one of those "the code needs more cleaning" issues. `includeCurrent` implies this is being run at Java crash time. We only want to run this code for the startup-time initialize.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java (hunk 2 above)
Comment (mrwillis21) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java]: Yeah, no harm in that.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java (hunk 1 above)
Comment (mrwillis21) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/core/CrashlyticsController.java]: Added a comment.

**Pull request merged**